File: rerun-reports/02-python/20260829-032449/results/key-vault/data-plane/python/crud/key-vault-dp-python-crud/python-azure-skills/azure-skill-mcp/generated-code/key_vault_secrets_crud.py
"""Create, read, update, delete, and purge an Azure Key Vault secret."""

import logging
import os
import sys
from uuid import uuid4

from azure.core.exceptions import (
    AzureError,
    ClientAuthenticationError,
    HttpResponseError,
    ResourceNotFoundError,
)
from azure.identity import DefaultAzureCredential
from azure.keyvault.secrets import SecretClient


INITIAL_VALUE = "initial-value"
UPDATED_VALUE = "updated-value"


def configure_logging() -> None:
    logging.basicConfig(
        level=logging.INFO,
        format="%(asctime)s %(levelname)s %(message)s",
    )


def require_key_vault_url() -> str:
    key_vault_url = os.getenv("AZURE_KEY_VAULT_URL")
    if not key_vault_url:
        raise ValueError(
            "AZURE_KEY_VAULT_URL is required, for example "
            "'https://my-vault.vault.azure.net'."
        )
    return key_vault_url


def run_crud(client: SecretClient, secret_name: str) -> None:
    logging.info("Creating secret %r.", secret_name)
    created_secret = client.set_secret(secret_name, INITIAL_VALUE)
    logging.info("Created secret version %s.", created_secret.properties.version)

    logging.info("Reading secret %r.", secret_name)
    retrieved_secret = client.get_secret(secret_name)
    logging.info(
        "Read secret version %s successfully.",
        retrieved_secret.properties.version,
    )

    logging.info("Updating secret %r.", secret_name)
    updated_secret = client.set_secret(secret_name, UPDATED_VALUE)
    logging.info(
        "Updated secret to version %s.",
        updated_secret.properties.version,
    )

    logging.info("Deleting secret %r.", secret_name)
    delete_poller = client.begin_delete_secret(secret_name)
    delete_poller.result()
    logging.info("Secret deletion completed.")

    logging.info("Purging deleted secret %r.", secret_name)
    client.purge_deleted_secret(secret_name)
    logging.info("Secret purge completed.")


def main() -> int:
    configure_logging()

    try:
        key_vault_url = require_key_vault_url()
    except ValueError as error:
        logging.error("%s", error)
        return 2

    secret_name = os.getenv("AZURE_KEY_VAULT_SECRET_NAME")
    if not secret_name:
        secret_name = f"python-crud-{uuid4().hex}"

    try:
        with (
            DefaultAzureCredential() as credential,
            SecretClient(vault_url=key_vault_url, credential=credential) as client,
        ):
            run_crud(client, secret_name)
    except ClientAuthenticationError:
        logging.exception(
            "Authentication failed. Sign in with a supported DefaultAzureCredential "
            "method or configure a managed identity."
        )
        return 1
    except ResourceNotFoundError:
        logging.exception(
            "The vault or secret was not found during the CRUD operation."
        )
        return 1
    except HttpResponseError:
        logging.exception(
            "Azure Key Vault rejected an operation. Verify the vault URL, RBAC "
            "permissions, firewall settings, and that purge protection is disabled."
        )
        return 1
    except AzureError:
        logging.exception("An Azure SDK error interrupted the CRUD operation.")
        return 1

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: rerun-reports/05-retries/07-js-ts-key-vault-baseline/20260829-192138/results/key-vault/data-plane/js-ts/crud/key-vault-dp-js-ts-crud/js-ts-azure-skills/baseline/generated-code/src/key-vault-secrets-crud.ts
import { DefaultAzureCredential } from "@azure/identity";
import { SecretClient } from "@azure/keyvault-secrets";

const secretName = "my-secret";

function getVaultUrl(): string {
  const vaultUrl = process.env.AZURE_KEY_VAULT_URL;

  if (!vaultUrl) {
    throw new Error(
      "AZURE_KEY_VAULT_URL is required (for example, https://my-vault.vault.azure.net).",
    );
  }

  try {
    return new URL(vaultUrl).toString();
  } catch {
    throw new Error("AZURE_KEY_VAULT_URL must be a valid URL.");
  }
}

function formatError(error: unknown): string {
  if (error instanceof Error) {
    const statusCode =
      "statusCode" in error && typeof error.statusCode === "number"
        ? ` (HTTP ${error.statusCode})`
        : "";
    return `${error.name}${statusCode}: ${error.message}`;
  }

  return String(error);
}

async function main(): Promise<void> {
  const credential = new DefaultAzureCredential();
  const client = new SecretClient(getVaultUrl(), credential);

  try {
    console.log(`Creating secret "${secretName}"...`);
    await client.setSecret(secretName, "my-secret-value");

    const createdSecret = await client.getSecret(secretName);
    if (createdSecret.value === undefined) {
      throw new Error(`Secret "${secretName}" was read without a value.`);
    }
    console.log(`Secret value: ${createdSecret.value}`);

    console.log(`Updating secret "${secretName}"...`);
    await client.setSecret(secretName, "updated-value");

    console.log(`Deleting secret "${secretName}"...`);
    const deletePoller = await client.beginDeleteSecret(secretName);
    await deletePoller.pollUntilDone();

    console.log(`Purging secret "${secretName}"...`);
    await client.purgeDeletedSecret(secretName);

    console.log("Secret CRUD operations completed successfully.");
  } catch (error: unknown) {
    console.error(`Key Vault operation failed: ${formatError(error)}`);
    process.exitCode = 1;
  }
}

void main();
